pkg/kubectl/cmd/rollout/rollout_history.go

```diff
@@ -98,27 +98,17 @@ func RunHistory(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, args []st
 		if err != nil {
 			return err
 		}
-		historyInfo, err := historyViewer.History(info.Namespace, info.Name)
+		historyInfo, err := historyViewer.ViewHistory(info.Namespace, info.Name, revision)
 		if err != nil {
 			return err
 		}
 
-		if revisionDetail > 0 {
-			// Print details of a specific revision
-			template, ok := historyInfo.RevisionToTemplate[revisionDetail]
-			if !ok {
-				return fmt.Errorf("unable to find revision %d of %s %q", revisionDetail, mapping.Resource, info.Name)
-			}
-			fmt.Fprintf(out, "%s %q revision %d\n", mapping.Resource, info.Name, revisionDetail)
-			kubectl.DescribePodTemplate(template, out)
-		} else {
-			// Print all revisions
-			formattedOutput, printErr := kubectl.PrintRolloutHistory(historyInfo, mapping.Resource, info.Name)
-			if printErr != nil {
-				return printErr
-			}
-			fmt.Fprintf(out, "%s\n", formattedOutput)
+		header := fmt.Sprintf("%s %q history viewed", mapping.Resource, info.Name)
```

Comment:
This seems redundant to me especially when `--revision` isn't specified. I'd expect it work more like `get` (i.e. prints the table without this header, we already have the column headers anyways).
```console
$ kubectl rollout history deployments 
deployments "nginx" history viewed
REVISION	CHANGE-CAUSE
1		<none>
2		<none>
```